### PR TITLE
Feature: Index pages instead of tt_content

### PR DESCRIPTION
**Pull request by DanielSiepmann**
As that makes much more sense. Still content is added to each page.
Also an abstract is added, where the fields used are configurable.

### CODE REVIEW

File: Classes/Domain/Index/AbstractIndexer.php [hunk 1]
```diff
@@ -30,6 +32,16 @@
...
 
     /**
+     * @var ConfigurationContainerInterface
+     */
+    protected $configuration;
+
+    /**
+     * @var string
+     */
+    protected $identifier;
```
Comment (justusmoroni) [inline Classes/Domain/Index/AbstractIndexer.php:43]: In other parts of code you initialized it with an empty string.
Shouldn't this be consistent?

File: Classes/Domain/Index/TcaIndexer/PagesIndexer.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        $this->connection = $connection;
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * @param array &$record
+     */
+    protected function prepareRecord(array &$record)
+    {
+        $possibleTitleFields = ['nav_title', 'tx_tqseo_pagetitle_rel', 'title'];
```
Comment (justusmoroni) [inline Classes/Domain/Index/TcaIndexer/PagesIndexer.php:60]: Should this be hardcoded?
We should move this to an configuration.

File: Configuration/TypoScript/constants.txt [hunk 3]
```diff
@@ -9,15 +9,9 @@ plugin {
...
-                # https://github.com/DanielSiepmann/search_core/issues/24 but
-                # should also be added, together with additionalWhereClause
-                # based on doktypes
-                tt_content {
-                    additionalWhereClause (
-                        pages.doktype NOT IN (3, 199)
-                        AND tt_content.CType NOT IN ('gridelements_pi1', 'list', 'div', 'menu', 'shortcut', 'search', 'login')
-                    )
+                pages {
+                    additionalWhereClause = pages.doktype NOT IN (3, 199, 6, 254, 255, 199)
```
Comment (justusmoroni) [inline Configuration/TypoScript/constants.txt:13]: `199` is in list multiple times.
Maybe sort it?

File: Tests/Functional/Fixtures/Indexing/IndexTcaTable.xml [hunk 4]
```diff
@@ -60,4 +60,43 @@
...
+    <tt_content>
+        <uid>9</uid>
+        <pid>1</pid>
+        <tstamp>1480686370</tstamp>
+        <crdate>1480686370</crdate>
+        <hidden>0</hidden>
+        <sorting>72</sorting>
+        <CType>div</CType>
+        <header>not indexed due to ctype</header>
+        <bodytext>this is the content of header content element that should not get indexed</bodytext>
```
Comment (justusmoroni) [inline Tests/Functional/Fixtures/Indexing/IndexTcaTable.xml:72]: What?

Fix in `Tests/Functional/Indexing/PagesIndexerTest.php` line 53 as well.

**Review by justusmoroni — CHANGES_REQUESTED** [review]

File: Classes/Domain/Index/TcaIndexer/PagesIndexer.php (hunk 2 above)
Comment (DanielSiepmann) [reply Classes/Domain/Index/TcaIndexer/PagesIndexer.php:60]: Yes, but for now this should be fine as two are core fields and one is just a supported extension. Once needed we can shift this to configuration with default setup.

File: Configuration/TypoScript/constants.txt (hunk 3 above)
Comment (DanielSiepmann) [reply Configuration/TypoScript/constants.txt:13]: Sorting is done based on page type grouping, not the identifier. I've removed the second 199.

File: Tests/Functional/Fixtures/Indexing/IndexTcaTable.xml (hunk 4 above)
Comment (DanielSiepmann) [reply Tests/Functional/Fixtures/Indexing/IndexTcaTable.xml:72]: Nothing to fix in test, as this content element is NOT indexed due to the CType, while the header element is indexed.

But of course I've fixed the content element type in bodytext.

**Pull request closed without merging**